### ISSUE TITLE
Fix proxy memory buffering: stream request/response bodies in backendProxy.ts

### ISSUE THREAD

**Issue by madara88645** [madara88645/Compiler#528]
## Problem

`web/lib/server/backendProxy.ts` currently buffers the entire request body and response body in memory before forwarding:

```ts
// request side — line ~81
body: isBodylessMethod(request.method) ? undefined : await request.arrayBuffer()

// response side — line ~48
await response.arrayBuffer()
```

On Fly.io with 512 MB RAM, large RAG payloads (uploads, search results) can trigger OOM kills. This was flagged in PR #512 before merge but was not addressed in PR #523.

## Expected fix

Pass the body as a `ReadableStream` instead of materializing it:

```ts
// request side
body: isBodylessMethod(request.method) ? undefined : request.body

// response side — forward the upstream response body directly
return new Response(upstreamResponse.body, {
  status: upstreamResponse.status,
  headers: filteredHeaders,
})
```

This keeps both sides streaming end-to-end, which is the standard Next.js proxy pattern.

## Files to change
- `web/lib/server/backendProxy.ts`
- `web/lib/server/backendProxy.test.ts` (update mocks accordingly)

## Acceptance criteria
- No `arrayBuffer()` call on either request or response path
- Existing proxy-routes tests still pass
- `npm run build` passes

**Comment by madara88645** [comment]
Resolved by #532: backendProxy now streams request and response bodies via ReadableStream, no arrayBuffer() in either path.
